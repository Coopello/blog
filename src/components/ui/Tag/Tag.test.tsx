import renderer from "react-test-renderer";

import { Tag } from "./";
// ______________________________________________________
//
describe("ui/Tag", () => {
  it("Snap Shot Chip", () => {
    const component = renderer.create(
      <Tag color="#E10718" text="front-end" filled />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Snap Shot outline Chip", () => {
    const component = renderer.create(<Tag color="#E10718" text="front-end" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Snap Shot button Chip", () => {
    const onClick = jest.fn();

    const component = renderer.create(
      <Tag color="#E10718" text="front-end" onClick={onClick} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
