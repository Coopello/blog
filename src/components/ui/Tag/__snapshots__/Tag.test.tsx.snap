// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Tag Snap Shot Chip 1`] = `
<span
  className="rounded-sm text-sm"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    {
      "backgroundColor": "#E10718",
      "color": "white",
      "opacity": 1,
      "paddingLeft": 12,
      "paddingRight": 12,
    }
  }
>
  #
  front-end
</span>
`;

exports[`ui/Tag Snap Shot button Chip 1`] = `
<button
  className="rounded-sm text-sm"
  onClick={[MockFunction]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    {
      "backgroundColor": "white",
      "color": "#E10718",
      "paddingLeft": 8,
      "paddingRight": 8,
    }
  }
>
  #
  front-end
</button>
`;

exports[`ui/Tag Snap Shot outline Chip 1`] = `
<span
  className="rounded-sm text-sm"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    {
      "backgroundColor": "white",
      "color": "#E10718",
      "paddingLeft": 8,
      "paddingRight": 8,
    }
  }
>
  #
  front-end
</span>
`;
